api/python/t4/bucket.py

```diff
@@ -27,6 +39,47 @@ def __init__(self, bucket_uri):
 
         self._uri = 's3://{}/'.format(bucket)
         self._bucket = bucket
+        self.config(quiet=True)
+
+    def config(self, config_url=CONFIG_URL, quiet=False):
+        """
+        Updates this bucket's search endpoint based on a federation config.
+        """
+        response = requests.get(config_url)
+        if not response.ok:
+            # just don't do anything
+            if not quiet:
+                raise QuiltException("Failed to configure bucket search endpoint")
+            return
+        configs = json.loads(response.text).get('configs', None)
+        if not configs:
+            if not quiet:
+                raise QuiltException("Config at config_url malformed")
+            return
+        if self._bucket in configs:
+            self._search_endpoint = configs[self._bucket]['search_endpoint']
+        elif not quiet:
+            raise QuiltException("Config info not found for this bucket")
+
+    def search(self, query):
+        """
+        Execute a search against the configured search endpoint.
+
+        query: query string to search
+
+        Returns either the request object (in case of an error) or
+                a list of objects with the following keys:
+            key: key of the object
+            version_id: version_id of object version
+            operation: Create or Delete
+            meta: metadata attached to object
+            size: size of object in bytes
+            text: indexed text of object
+            source: source document for object (what is actually stored in ElasticSeach)
+            time: timestamp for operation
+
+        """
+        return search(query, self._search_endpoint)
```

Comment:
We're silently placing the user in a scenario where the bucket they are interacting with may or may not be searchable. If they later go to try to search that bucket, and that bucket isn't searchable, they get some obscure error---frustrating.

We should throw a better error when search fails because the bucket isn't properly configured for it.

Reply:
I made the error messages a little better, what do you think?

Reply:
Current error message:

> QuiltException: Config info not found for this bucket at config_url

I would shorten this to just:

> QuiltException: Config info not found for this bucket

"at `config_url`" implies to me that there is `Bucket.config_url` attribute that I can access to further debug the issue, when in reality no such attribute exists, which is unhelpful.

Once we have documentation we'll need a section for FAQs or common issues that contains an entry for this config problem, which we'll probably want to point the user to from this error message. Task for the future.